Bio/phenotype/PhenMicro.py

```diff
@@ -0,0 +1,1126 @@
+
+        A new PlateRecord object is returned, having the same id as the
+        left operand
+        """
+        if not isinstance(plate, PlateRecord):
+            raise TypeError('Expecting a PlateRecord object')
+
+        if set([x.id for x in self]) != set([x.id for x in plate]):
+            raise ValueError('The two plates have different wells')
+
+        wells = []
+
+        for w in self:
+            wells.append(w - plate[w.id])
+
+        newp = PlateRecord(self.id, wells=wells)
+
+        return newp
+
+    def get_row(self, row):
+        """Get all the wells of a given row
+
+        A row is identified with a letter (e.g. 'A')
+        """
+        # Key is casted to str implicitly
+        try:
+            row = str(row)
+        except Exception:
+            # Is it even possible to get an exception here?
+            raise ValueError('Row identifier should be string-like')
+        if len(row) > 1:
+            raise ValueError('Row identifier must be of maximum one letter')
+
+        for w in sorted(filter(lambda x: x.startswith(row), self._wells)):
+            yield self._wells[w]
+
+    def get_column(self, column):
+        """Get all the wells of a given column
+
+        A column is identified with a number (e.g. '6')
+        """
+        # Column is casted to int implicitly
+        try:
+            column = int(column)
+        except Exception:
+            raise ValueError('Column identifier should be a number')
+
+        # A 96-well plate has well numbers in two digits
+        for w in sorted(filter(lambda x: x.endswith('%02d' % column),
+                               self._wells)):
+            yield self._wells[w]
+
+    def subtract_control(self, control='A01', wells=None):
+        """Subtract a 'control' well from the other plates wells
+
+        By default the control is subtracted to all wells, unless
+        a list of well ID is provided
+
+        The control well should belong to the plate
+        A new PlateRecord object is returned
+        """
+        if control not in self:
+            raise ValueError('Control well not present in plate')
+        wcontrol = self[control]
+
+        if wells is None:
+            wells = self._wells.keys()
+
+        missing = set()
+        for w in wells:
+            if w not in self:
+                missing.add(w)
+        if len(missing) > 0:
+            raise ValueError('Some wells to be subtracted are not present')
+
+        nwells = []
+
+        for w in self:
+            if w.id in wells:
+                nwells.append(w - wcontrol)
+            else:
+                nwells.append(w)
+
+        newp = PlateRecord(self.id, wells=nwells)
+
+        return newp
+
+    def __repr__(self):
+        """Returns a (truncated) representation of the plate for debugging."""
+        if len(self._wells) > 5:
+            # Show the last well and the first four
+            return "%s('%s...%s')" % (self.__class__.__name__,
```

Comment:
How do you feel about:

```
PlateRecord('WellRecord['A01'], WellRecord['A02'], WellRecord['A03'], WellRecord['A04']...WellRecord['H12']')
```

vs:

```
PlateRecord('WellRecord['A01'], WellRecord['A02'], WellRecord['A03'], WellRecord['A04'], ..., WellRecord['H12']')
```

I've tried to find anything in the Python style guidelines but my Google searching was unsuccessful.

Also maybe first 3 and last 1 would be enough?